docs/workflow.md

```diff
@@ -0,0 +1,49 @@
+# Workflow
```

Comment:
```suggestion
# NuGet.Client Development Workflow
```

Reply:
Similar to the above, I don't see the point in blasting the repo name everywhere :)